spec.bs

```diff
@@ -116,6 +119,43 @@ if (document.wasDiscarded) {
 }
 </pre>
 
+Feature Policies {#feature-policies}
+=====================
+To improve CPU utilization of sub-frames that are not critical to a page's usability two feature policies are defined:
+ * execution-while-not-rendered, which has a [=default allowlist=] of '*'
+ * execution-while-out-of-viewport, which has a [=default allowlist=] of '*'
+
+<dfn export>execution-while-not-rendered</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for frames that are not part of rendering. eg. `display: none`.
+
+<dfn export>execution-while-out-of-viewport</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for frames that are scrolled out of the current visible viewport.
+
+Application of these policies is accomplished by placing the frame (and its decendants) in a [=/frozen=] state when the following
+conditions have been met:
+ * [=iframe load event steps=] have been run
+ * the policy is disabled for the frame
+ * relevant policy condition applies (not rendered or scrolled out of view)
+
+If these conditions aren't met the frame should be in the [=/unfrozen=] state.
+
+<pre class="example highlight">
+
+// The subframe will be frozen immediately after it is loaded.
+
+&lt;iframe allow="execution-while-not-rendered 'none'"
+  src="subframe.html" style="display:none"&gt;&lt;/iframe&gt;
+</pre>
+
+To run the <dfn export>update frame throttling steps</dfn> run the following steps:
+1. If [=iframe-load-in-progress=] is set, return.
```

Comment:
This needs to reference a specific _document_'s iframe load in progress flag. Probably the document should be an argument to the "update frame throttling steps"?